### PR TITLE
[docs] REST API documentation

### PR DESCRIPTION
**Pull request by arzoo14**
This PR contains the REST API documentation alongwith the basic setup and some necessary changes in UAG and PG books.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -171,3 +193,7 @@
 
 # A list of files that should not be packed into the epub file.
 epub_exclude_files = ['search.html']
+
+from docutils.parsers.rst import directives
+from sphinx.directives.code import CodeBlock
+directives.register_directive('code', CodeBlock)
```
Comment (andreasgerstmayr) [inline docs/conf.py:199]: Was this required for the ``` blocks in the openapi.yaml file?
If yes, is it possible to change the syntax highlighting language? Some code blocks look weird, for example https://demozxc.readthedocs.io/en/latest/api/#operation/get/series/instances (`605` is colored differently than the rest of the ID).

File: docs/specs/openapi.yaml [hunk 2]
```diff
@@ -0,0 +1,948 @@
...
+    The timeout interval is configurable at context creation time, and as such the *polltime* parameter can be used anywhere the *hostspec* is specified. It sets the context timeout in terms of length of inactive time. The unit for the timeout value is seconds and the default is 5.
+
+    To specify a specific existing context in any PMAPI web request, the endpoints can be accessed with either the *context* parameter or enbedded in the endpoint URL itself, such as **/pmapi/[number]/fetch**.
+paths:
+  /metrics:
+    get:
+      tags:
+      - OPEN METRICS
+      summary: Fetches current values and metadata 
+      description: " `Get metrics` fetches current values and metadata for all metrics, or only metrics\
```
Comment (andreasgerstmayr) [inline docs/specs/openapi.yaml:68]: Can you format this block with `description: |` and normal line breaks (like in the other API methods)?

File: docs/specs/openapi.yaml [hunk 3]
```diff
@@ -0,0 +1,948 @@
...
+
+
+
+
+
+
+
+components:
+  schemas:
+    response1:
```
Comment (andreasgerstmayr) [inline docs/specs/openapi.yaml:762]: Can you give them names? For example "seriesIds"

File: docs/specs/openapi.yaml [hunk 4]
```diff
@@ -0,0 +1,948 @@
...
+      title: Without Time Window
+      type: array
+      items:
+        type: string
+      example:
+        - 9d8c7fb51ce160eb82e3669aac74ba675dfa8900
+        - ddff1bfe286a3b18cebcbadc1678a68a964fbe9d
+        - 605fc77742cd0317597291329561ac4e50c0dd12
+
+    response2:
```
Comment (andreasgerstmayr) [inline docs/specs/openapi.yaml:772]: This would be `seriesValues`

File: docs/specs/openapi.yaml [hunk 5]
```diff
@@ -0,0 +1,948 @@
...
+            domainname: acme.com
+            groupid: 1000
+            hostname: www.acme.com
+            latitude: -25.28496
+            longitude: 152.87886
+            machineid: 295b16e3b6074cc8bdbda8bf96f6930a
+            platform: dev
+            userid: 1000
+                
+    response4:
```
Comment (andreasgerstmayr) [inline docs/specs/openapi.yaml:840]: `labelNames`

File: docs/specs/openapi.yaml [hunk 6]
```diff
@@ -0,0 +1,948 @@
...
+        properties:
+          series:
+            type: string
+          name:
+            type: string
+      example:
+        - series: 605fc77742cd0317597291329561ac4e50c0dd12
+          name: disk.dev.read_bytes
+                
+    response7:
```
Comment (andreasgerstmayr) [inline docs/specs/openapi.yaml:891]: `metricNames`

**Review by andreasgerstmayr — CHANGES_REQUESTED** [review]
Looks great! I've added a few comments.

File: docs/conf.py (hunk 1 above)
Comment (arzoo14) [reply docs/conf.py:199]: Ah! I forgot to remove these lines. These were just to check something. 
Yes, I have changed the syntax highlighting language to `bash`

File: docs/specs/openapi.yaml (hunk 2 above)
Comment (arzoo14) [reply docs/specs/openapi.yaml:68]: Done!

File: docs/specs/openapi.yaml (hunk 3 above)
Comment (arzoo14) [reply docs/specs/openapi.yaml:762]: Done!

File: docs/specs/openapi.yaml (hunk 4 above)
Comment (arzoo14) [reply docs/specs/openapi.yaml:772]: Done!

File: docs/specs/openapi.yaml (hunk 5 above)
Comment (arzoo14) [reply docs/specs/openapi.yaml:840]: Done!

File: docs/specs/openapi.yaml (hunk 6 above)
Comment (arzoo14) [reply docs/specs/openapi.yaml:891]: Done!

**Pull request merged**